app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -101,22 +133,26 @@ def parse_az(az)
       end
 
       def parse_instance(instance)
-        uid         = "#{@subscription_id}\\#{instance.fetch_path('resourceGroup')}\\#{instance.fetch_path('name')}"
+        uid = resource_uid(@subscription_id,
+                           instance.fetch_path('resourceGroup'),
+                           instance.fetch_path('type').downcase,
+                           instance.fetch_path('name'))
```

Comment:
@bzwei  - If an instance is deleted and later a new instance is created with the same name in the same resource group for the same subscriber I think it will not be treated as a new instance if this UID is used.
Can you map with a GUID somehow instead?